CS.API/BackgroundServices/BackgroundNotificationService.cs

```diff
@@ -43,6 +43,8 @@ private async Task CheckUnReadMessages()
         }
         
         var tasks = messages.Select(emailService.SendEmail);
+        
+        //TODO: save changes seems to be missing here
```

Comment:
maybe I'm missing something, but it seems like you do not save changes here. Do you do that somewhere outside of this method?